src/main/java/org/broadinstitute/hellbender/engine/ReadsDataSourcePool.java

```diff
@@ -57,14 +58,19 @@ public void close() {
     private static class Factory extends BasePoolableObjectFactory<ReadsPathDataSource> {
 
         private final List<Path> paths;
+        private final SamReaderFactory factory;
 
-        private Factory(final List<Path> paths) {
+        private Factory(final List<Path> paths, final Path referencePath) {
+            this.factory = SamReaderFactory.makeDefault();
+            if (referencePath != null) {
+                this.factory.referenceSequence(referencePath);
+            }
```

Comment:
I agree... My intention is to fix this code before the release lest it cause problems when people start using DRAGEN-GATK. I agree that this is potentially missing other engine level features and in reality I suspect the best solution for threading this will be to reimplement it in spark rather than dealing with this code. That will have to happen at another time.